src/components/test-slot/test-slot/test-slot.ts

```diff
@@ -83,11 +92,15 @@ export class TestSlotComponent implements SlotComponent, OnInit {
     this.componentState = {
       testStatus$: this.store$.pipe(
         select(getTests),
-        select((tests) => getTestStatus(tests, slotId)),
+        select((tests) => this.derivedTestStatus || getTestStatus(tests, slotId)),
       ),
       testActivityCode$: this.store$.pipe(
         select(getTests),
-        map((tests) => getActivityCodeBySlotId(tests, this.slot.slotDetail.slotId)),
+        map((tests) => this.derivedActivityCode || getActivityCodeBySlotId(tests, this.slot.slotDetail.slotId)),
```

Comment:
We could use the ```slotId``` that has been de-structured on first line of  ```ngOnInit``` and same below on 103